google-cloud-firestore/src/main/java/com/google/cloud/firestore/Transaction.java

```diff
@@ -78,11 +85,18 @@ Transaction wrapResult(ApiFuture<WriteResult> result) {
     BeginTransactionRequest.Builder beginTransaction = BeginTransactionRequest.newBuilder();
     beginTransaction.setDatabase(firestore.getDatabaseName());
 
-    if (previousTransactionId != null) {
+    if (TransactionOptionsType.READ_WRITE.equals(transactionOptions.getType())
+        && previousTransactionId != null) {
       beginTransaction
           .getOptionsBuilder()
           .getReadWriteBuilder()
           .setRetryTransaction(previousTransactionId);
+    } else if (TransactionOptionsType.READ_ONLY.equals(transactionOptions.getType())
+        && transactionOptions.getReadTime() != null) {
+      beginTransaction
+          .getOptionsBuilder()
+          .getReadOnlyBuilder()
+          .setReadTime(transactionOptions.getReadTime());
```

Comment:
I think we must call `beginTransaction.getOptionsBuilder().getReadOnlyBuilder()` for read-only transaction even if we don't have a read time. You can probably just drop ` && transactionOptions.getReadTime() != null` (and maybe also `&& previousTransactionId != null`).

Reply:
Good catch. I did need to handle the else branch here that was previously taken care of by the options class. I've updated the code to handle both cases.